sdk/scheduler/src/main/java/com/mesosphere/sdk/api/ArtifactResource.java

```diff
@@ -0,0 +1,130 @@
+package com.mesosphere.sdk.api;
+
+import com.mesosphere.sdk.config.ConfigStore;
+import com.mesosphere.sdk.config.ConfigStoreException;
+import com.mesosphere.sdk.offer.ResourceUtils;
+import com.mesosphere.sdk.specification.ConfigFileSpec;
+import com.mesosphere.sdk.specification.PodSpec;
+import com.mesosphere.sdk.specification.ServiceSpec;
+import com.mesosphere.sdk.specification.TaskSpec;
+import com.mesosphere.sdk.storage.StorageError.Reason;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+/**
+ * A read-only API for accessing file artifacts (e.g. config templates) for retrieval by executors.
+ */
+@Path("/v1/artifacts")
+public class ArtifactResource {
+    private static final String ARTIFACT_URI_FORMAT =
+            "http://api.%s.marathon." + ResourceUtils.VIP_HOST_TLD + "/v1/artifacts/template/%s/%s/%s/%s";
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    private final ConfigStore<ServiceSpec> configStore;
+
+    /**
+     * Returns a valid URL for accessing a config template artifact from a service task.
+     * Must be kept in sync with {@link #getTemplate(String, String, String, String)}.
+     */
+    public static String getTemplateUrl(
+            String serviceName, UUID configId, String podType, String taskName, String configName) {
+        return String.format(ARTIFACT_URI_FORMAT, serviceName, configId, podType, taskName, configName);
+    }
+
+    public ArtifactResource(ConfigStore<ServiceSpec> configStore) {
+        this.configStore = configStore;
+    }
+
+    /**
+     * Produces the content of the requested configuration template, or returns an error if that template doesn't exist
+     * or the data couldn't be read.
+     */
+    @Path("/template/{configurationId}/{podType}/{taskName}/{configurationName}")
+    @GET
+    public Response getTemplate(
```

Comment:
Can you please update the javadoc with the parameters to this method?  All are obvious to me except `configurationId`.